### ISSUE TITLE
interface for quality flag filters that matches discard/resample capabilities in core

### ISSUE THREAD

**Issue by wholmgren** [SolarArbiter/solarforecastarbiter-dashboard#367]
https://github.com/SolarArbiter/solarforecastarbiter-core/pull/580 adds new keys ``discard_before_resample`` and ``resample_threshold_percentage``. The dashboard should expose these keys to the user. It should also allow users to create multiple `QualityFlagFilter` specs comprised of multiple `quality_flags` (`QualityFlagFilter(['NIGHTTIME', 'USER FLAGGED'])` has a different behavior than `QualityFlagFilter(['NIGHTTIME']), QualityFlagFilter(['USER FLAGGED'])`). 

But, we're also talking about compressing all of the filters (quality, time, and value) into a single `Filter` object, so ok to wait on this until that discussion is settled.

**Comment by lboeman** [comment]
Sounds like we're going to need some sort of dynamic form for this, something like what we have for cost. Probably something we want to validate on the API as well.

**Comment by wholmgren** [comment]
Yes, I forgot to mention I thought it would copy some of the cost metrics configuration pattern. But I think we should do #368 first.

**Comment by lboeman** [comment]
This issue has uncovered an issue with the report cloning interface. The dashboard parses checked boxes on the front end and converts them to filter objects in the report parameters that look like:
```
{
    "quality_flags": [flag],
    "discard_before_resample": flag in core.validation.quality_mapping.DISCARD_BEFORE_RESAMPLE,
    "resample_threshold_percentage": 10
}
```
This is great when we want to supply sane defaults for users, but if the user is cloning a report made through the api that has more complex filtering configuration, we strip out everything but the quality flags and recreate from defaults as described as above.

**Comment by lboeman** [comment]
Since order matters here, do we want the interface to allow users to define and reorder these filters?

**Comment by wholmgren** [comment]
Maybe I'm missing something, but I don't think order matters. Grouping does matter though.
